src/components/Friend/EditMemberComponent.js

```diff
@@ -1,60 +1,107 @@
+  const groupID = groupInfo.id;
+  const itemID = item.id;
 
-const EditMemberComponent = ({ items, isMember, groupID }) => {
-  const userID = firebase.auth().currentUser.uid;
   const alertTitle = isMember
     ? "This user will be removed from the group"
     : "This invitation will be removed";
-  const onEditButtonPress = (item) => {
-    Alert.alert(
-      alertTitle,
-      "This action is irreversible. Do you want to continue?",
-      [
-        {
-          text: "Cancel"
-        },
-        {
-          text: "Continue",
-          onPress: () => {
-            if (isMember) {
-              removeUserFromGroup(groupID, item.id);
-            } else {
-              cancelGroupInvitation(groupID, item.id);
-            }
-          }
-        }
-      ]
-    );
-  };
 
-  const EditButton = ({ item }) => (
-    <Button
-      title="Remove"
-      buttonStyle={styles.removeButton}
-      icon={{ name: "delete", color: "white" }}
-      onPress={() => onEditButtonPress(item)}
-    />
-  );
+  const isCurrentUserOwner = currentID === groupInfo.owner;
+  const isVisible =
+    isExpanded &&
+    (!isMember || (!item.isOwner && (isCurrentUserOwner || !item.isAdmin)));
+
+  const buttonDetails = [
+    {
+      title: "Remove",
+      type: "material-community",
+      icon: "account-remove",
+      color: "red",
+      onPress: onRemovePress
+    }
+  ];
+
+  if (isCurrentUserOwner && isMember) {
+    buttonDetails.push({
+      title: item.isAdmin ? "Demote" : "Promote",
+      type: "material-community",
+      icon: item.isAdmin ? "account-arrow-down" : "account-arrow-up",
+      color: "blue",
+      onPress: onAdminUpdatePress
+    });
+  }
+
+  function onRemovePress() {
+    const handleRemove = () => {
+      isMember
+        ? removeUserFromGroup(groupID, itemID)
+        : cancelGroupInvitation(groupID, itemID);
+    };
 
-  return items.map((item, idx) => (
-    <ListItem.Swipeable
-      key={idx}
+    return Caution(alertTitle, handleRemove);
+  }
+
+  async function onAdminUpdatePress() {
+    if (item.isAdmin) {
+      await demoteAdminToMember(groupID, itemID);
+    } else {
+      await promoteMemberToAdmin(groupID, itemID);
+    }
+  }
+
+  const RenderButtons = () => {
+    if (!isVisible) {
+      return null;
+    }
+
+    return buttonDetails.map((detail, index) => (
+      <ListItem key={index} onPress={detail.onPress}>
+        <Icon
+          type={detail.type}
+          name={detail.icon}
+          size={30}
+          color={detail.color}
+        />
+        <ListItem.Content>
+          <ListItem.Title>{detail.title}</ListItem.Title>
+        </ListItem.Content>
+      </ListItem>
+    ));
+  };
+
+  return (
+    <ListItem.Accordion
       bottomDivider
-      rightContent={item.id !== userID && <EditButton item={item} />}
+      content={<ContactBar type={contactType.USER} item={item} />}
+      isExpanded={isExpanded}
+      onPress={changeExpanded}
```

Comment:
I think you can differentiate between admins & normal members because for admins the drop down does not show anything anyway